src/sagemaker/workflow/pipeline.py

```diff
@@ -515,6 +539,51 @@ def update_args(args: Dict[str, Any], **kwargs):
             args.update({key: value})
 
 
+class ImmutablePipeline(Pipeline):
+    """ImmutablePipeline to support pipelines that should be immutable."""
+
+    def update(
+        self,
+        role_arn: str,
+        description: str = None,
+        parallelism_config: ParallelismConfiguration = None,
+    ):
+        """Prevents an update on an ImmutablePipeline in the Workflow service.
+
+        Args:
+            role_arn (str): The role arn that is assumed by pipelines to create step artifacts.
+            description (str): A description of the pipeline.
+            parallelism_config (Optional[ParallelismConfiguration]): Parallelism configuration
```

Comment:
Nit: We don't need to add Optional here but we should mention (default value: None/or others) in docstring for each parameter with default value.
Plus we need to add Optional in the line 549